### ISSUE TITLE
05 Sept Meeting Tasklist

### ISSUE THREAD

**Issue by keeferrourke** [ian-james/IFS#112]
## Sign up system revisions
- [x] Restrict sign-up to *@uoguelph.ca domains
- [x] Look into LDAP Authentication via passport.js
    + If this will break anything, DO NOT implement LDAP
- [x] Do not provide experiment consent form on first login
    + Possibly direct users to tutorial-type resources so they can familiarize themselves with the system; prompt for experiment opt-in in the file-upload document
    + Experiment document may be subject to revisions pending ethics approval, so displaying it right away on log-in might not be a good idea.

## Skills revisions
- [x] Add better descriptions for skill rating on /skills
   + It is not clear that the skills are self-assessed by the numeric input
- [x] Consider adding a visual model for skill self-rating over time

### Feedback rating
- [x] Move feedback rating to top of sidebar
- [x] s/User Feedback/Rate this feedback
- [x] s/User Rating/Is this useful?

## File submission modal design
- [x] If user is not opted into experiment, show prompt to opt-in rather than displaying nothing
    + Make it clear that opting into the experiment will require no extra effort on their part

## Tutorials / Feature Discovery
- [x] Implement occasional popups/tooltips like "Did you know you can do &lt;feature X&gt;? Go to &lt;route X&gt; to try it!"
    + I think this would be best shown in the file submission model, in place of surveys every now and then
    + These messages can get annoying, so provide an opt-out for feature discovery (I will attach a UI mockup in another comment on this thread)
- [ ] Create /tutorials route which will show how to do things throughout the IFS with screenshots, descriptions, and screencast embeds
    + Add this route to the IFS footer, grouped near the Wiki link
- [ ] Create screencasts of IFS features and upload to SOCS youtube channel

## Tools revision
- [x] Make c99 the default language standard for gcc/clang
- [x] Add gnu* standards as options
- [x] Better explain the differences between gcc/clang in their tool descriptions.

### [APA Checker](https://github.com/keeferrourke/pyapa)
- [ ] Needs input from other computer scientists and psychologists to develop more regexes

**Comment by keeferrourke** [comment]
For each of the above task groups, it may be easier to discuss in separate issues.

**Comment by ian-james** [comment]
I would like to prioritize this into categories based on what must be done before introduction and what can be updated once active.

MUST: We have to make sure we have the resources to succeed.  Lets both focus on this for the week. We need real numbers from multiple machines here. Let research and find out how to benchmark this and include our own numbers in a shared doc.

1) Before Public Release:

-   The minor UI tweaks, text improvements, descriptions, and tool revisions. Lets keep these to tasks that take 10 minutes or less.

- LDAP and consent form removal.

2) Tooltip feature information and additional videos or tutorials   support .


Taking uoguelph requirement out that should be handled by ldap.

**Comment by ian-james** [comment]
@uoguelph.ca mandatory

**Comment by ian-james** [comment]
Modified opt-in prompt to be part of the tips.

**Comment by ian-james** [comment]
Essentially complete except tutorials,videos, and screencast.
Will be reviewing IFS until Thursday release.

**Comment by keeferrourke** [comment]
If we can get the SOCS YouTube credentials, I'll happily make the tutorials available. Can you poke Judi? She hasn't responded to any of my requests for credentials... (not that I blame her, she should enjoy her vacation 😄 )

**Comment by ian-james** [comment]
I always thought of those as a nice to have.  I'll poke Judi at some point about videos. We could always post them ourselves on YouTube under GuelphLabs and then transfer them to Judi too.

**Comment by keeferrourke** [comment]
@ian-james Have you heard anything from the other guys, re: improving the APA checker? I plan to take a look at it today and hopefully get some of the problematic patterns fixed.

**Comment by ian-james** [comment]
Created a separate issue for the last remaining videos
